src/T2DMap.cpp

```diff
@@ -3079,9 +3082,57 @@ void T2DMap::mousePressEvent(QMouseEvent* event)
         }
         connect(mapper, SIGNAL(mapped(QString)), this, SLOT(slot_userAction(QString)));
     }
+
+    toggleSelectionWidget();
     update();
 }
 
+void T2DMap::toggleSelectionWidget() {
```

Comment:
This name is confusing - `updateSelectionWidget` or `regenerateSelectionWidget` would be more appropriate IMHO.

I think there is a further place - in the `mouseMoveEvent(...)` in:
```diff
             default:
                 getCenterSelection(); // Sets mMultiSelectionHighlightRoomId to (a) central room
             }

-            if (mMultiSelectionSet.size() > 1) {
-                // We don't want to cause calls to slot_roomSelectionChanged() here!
- // A whole chunk of code here that is equivalent to:
+            toggleSelectionWidget();
```
where this method can be called though you might need to reposition the `update()` call at the bottom of this one:
```diff
         mMultiSelectionListWidget.selectAll();
         mMultiSelectionListWidget.blockSignals(false);
         mMultiSelectionListWidget.show();
-        update();
     } else {
         mMultiSelectionListWidget.hide();
     }
+    update();
 }
```

Reply:
Except this method really toggles selection widget based on state of selection.
But if you find it confusing no probs `updateSelectionWidget` is fine as well.
Actually `update()` is called right after this method call, so in matter of fact, but since it schedules repaint, I guess it can be called multiple times - therefore I've moved in to the bottom, so method can be called without worrying to call update externally.